nulls.ipynb

```diff
@@ -0,0 +1,288 @@
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {
+    "colab": {
+     "base_uri": "https://localhost:8080/"
+    },
+    "id": "QX7HnLig42Ya",
+    "outputId": "1a16bbc3-c725-4eb0-9c4c-3f454388cc39"
+   },
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "<pyarrow.lib.DoubleArray object at 0x7fe64d8236a0>\n",
+       "[\n",
+       "  1.2,\n",
+       "  3.4,\n",
+       "  9,\n",
+       "  null,\n",
+       "  2.9\n",
+       "]"
+      ]
+     },
+     "execution_count": 3,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "arr = pa.array([1.2, 3.4, 9.0, None, 2.9])\n",
+    "arr"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {
+    "colab": {
+     "base_uri": "https://localhost:8080/"
+    },
+    "id": "NNcrvbyv7cvN",
+    "outputId": "23ec4c17-0f51-46ee-b7f7-b745fbf5eab8"
+   },
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "<nanoarrow.c_lib.CArrayView>\n",
+       "- storage_type: 'double'\n",
+       "- length: 5\n",
+       "- offset: 0\n",
+       "- null_count: 1\n",
+       "- buffers[2]:\n",
+       "  - validity <bool[1 b] 11101000>\n",
+       "  - data <double[40 b] 1.2 3.4 9.0 0.0 2.9>\n",
+       "- dictionary: NULL\n",
+       "- children[0]:"
+      ]
+     },
+     "execution_count": 4,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "na.c_array_view(arr)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "Nc7uYpry5u0u"
+   },
+   "source": [
+    "**Attention**: Arrow uses [least-significant bit (LSB) numbering](https://en.wikipedia.org/wiki/Bit_numbering) (also known as bit-endianness). This means that within a group of 8 bits, we read right-to-left. However, the `nanoarrow` repr of the validity buffer in the example above already takes that into account and shows the values in logical order matching the position in the array. The diagram above shows it as how it is actually stored in memory."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {
+    "colab": {
+     "base_uri": "https://localhost:8080/"
+    },
+    "id": "DuIwrjV7Vqlb",
+    "outputId": "394ceb73-cf0e-4c91-f216-aa613ff3adca"
+   },
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "b'\\x17'"
+      ]
+     },
+     "execution_count": 5,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "validity_bitmap_buffer = arr.buffers()[0]\n",
+    "validity_bitmap_buffer.to_pybytes()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "In this case of a small array of 5 values, the validity bitmap consists of a single byte:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "array([23], dtype=uint8)"
```

Comment:
This part where bits are read as bytes and `uint8` is a bit confusing for me 😊 Can we use only `np.unpackbits` or maybe add a bit more information?